src/gpuarray_array_collectives.c

```diff
@@ -0,0 +1,108 @@
+#include "gpuarray/array.h"
+#include "gpuarray/buffer_collectives.h"
+#include "gpuarray/collectives.h"
+#include "gpuarray/error.h"
+
+#include "private.h"
+
+static inline int find_total_elems(const GpuArray* array)
+{
```

Comment:
I know, but I would prefer to hide that specific detail inside the API in case they ever decide to keep up with reality and switch to size_t.

Otherwise this limits the size that we can handle to 4G and we would have to expose a new API to handle 64-bits later.